editor/src/components/editor/actions/actions.tsx

```diff
@@ -4873,36 +4875,88 @@ export const UPDATE_FNS = {
     if (targetElementCoords != null && isFiniteRectangle(targetElementCoords)) {
       const isNavigatorOnTop = !editor.navigator.minimised
       const containerRootDiv = document.getElementById('canvas-root')
-      if (action.keepScrollPositionIfVisible && containerRootDiv != null) {
-        const containerDivBoundingRect = containerRootDiv.getBoundingClientRect()
-        const navigatorOffset = isNavigatorOnTop ? LeftPaneDefaultWidth : 0
+      const navigatorOffset = isNavigatorOnTop ? LeftPaneDefaultWidth : 0
+
+      // This returns the offset for 'to-origin' scroll behaviours, or used as the default
+      // for the other behaviours.
+      function canvasOffsetToOrigin(frame: CanvasRectangle): CanvasVector {
+        const baseCanvasOffset = isNavigatorOnTop ? BaseCanvasOffsetLeftPane : BaseCanvasOffset
+        return Utils.pointDifference(frame, baseCanvasOffset)
+      }
+
+      function canvasOffsetToCenter(
+        frame: CanvasRectangle,
+        bounds: CanvasRectangle | null,
+      ): CanvasVector {
+        if (bounds == null) {
+          return canvasOffsetToOrigin(frame) // fallback default
+        }
+        const scale = 1 / editor.canvas.scale
+        const canvasCenter = getRectCenter(
+          canvasRectangle({
+            x: bounds.x,
+            y: bounds.y,
+            width: bounds.width * scale,
+            height: bounds.height * scale,
+          }),
+        )
+        const topLeftTarget = canvasPoint({
+          x: canvasCenter.x - frame.width / 2 - bounds.x + (navigatorOffset / 2 + 10) * scale,
```

Comment:
What's with the `+ 10`?

Reply:
ah it's an ugly remnant of the navigator padding (20 / 2), I updated to use `DefaultNavigatorWidth` instead of `LeftPaneDefaultWidth` which correctly is 280 instead of 260 https://github.com/concrete-utopia/utopia/pull/3870/commits/f5d3088ea55fdc0a74197b571c2aaa5783721126